### PR TITLE
Allow JsonType to store any json-compatible type

### PR DESCRIPTION
**Pull request by simPod**
Resolves https://github.com/phpstan/phpstan-doctrine/issues/194

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
This needs a test. You can see how other descriptor changes are tested in commits that changed the same directory: https://github.com/phpstan/phpstan-doctrine/commits/master/src/Type/Doctrine/Descriptors

**Comment by ondrejmirtes** [comment]
Please test the changes locally before spamming the CI, thanks.

**Comment by simPod** [comment]
I'm used to leverage CI instead of my local hw so this workflow approvals are breaking my workflow. sorry for that tho.

File: src/Type/Doctrine/Descriptors/JsonType.php [hunk 1]
```diff
@@ -15,12 +24,30 @@ public function getType(): string
...
-		return new \PHPStan\Type\ArrayType(new MixedType(), new MixedType());
+		return new \PHPStan\Type\UnionType([
+			new ArrayType(new MixedType(), new MixedType()),
+			new BooleanType(),
+			new FloatType(),
+			new IntegerType(),
+			new NullType(),
+			new ObjectType(JsonSerializable::class),
+			new ObjectType(stdClass::class),
+			new StringType(),
```
Comment (ondrejmirtes) [inline src/Type/Doctrine/Descriptors/JsonType.php:49]: Some DRY would be nice here. Also reuse the same `MixedType` instance.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Review by ondrejmirtes — APPROVED** [review]

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**